index.html

```diff
@@ -118,16 +205,62 @@
     // TRANSFORM
     txData = data.slice();
 
-    if (options.yearRange !== 'all') {
-      var yearRange = options.yearRange.split('-');
+    if (options.SOCGroup !== 'all') {
+      var SOCGroup=options.SOCGroup;
       txData = txData.filter(function (d) {
-        return d.year >= yearRange[0] && d.year <= yearRange[1];
+        return d.SuperGroup === SOCGroup;
       });
     }
 
+
     // SCATTERPLOT
 
+    var points=svg.selectAll('.point')
+      .data(txData)
+
+      points.enter().append('circle')
+      .attr('class', 'point')
+      .attr('r', 5)
+
+      points
+      .attr('cx', function(d) { return x(d.WF_Female); })
+      .attr('cy',  function(d) { return y(d.MedEarn); })
+
+
+      points.exit().remove();
+
+
     // HISTOGRAMS
+    var map = data.map(function(d){return d.WF_Female;})
+
+    var histogram = d3.layout.histogram()
+                      .bins(x.ticks())(map);
+
+    var bars = svg.selectAll(".bar")
+                  .data(histogram)
+                  .enter();
```

Comment:
So compare and contrast this with `points` above. When we immediately call `.enter()`, we're changing what gets assigned to the `bars` variable. Now it's just the enter selection. This is why it's never changes after the first time it's called — the enter selection is always empty because we don't need to add any new bars.